test/Microsoft.TemplateEngine.IDE.IntegrationTests/End2EndTests.cs

```diff
@@ -300,5 +300,36 @@ internal async Task Test_DryRunAsync_OnInvalidParamsPassed()
             Assert.Equal(CreationResultStatus.InvalidParamValues, result.Status);
             Assert.Equal("userPort1, userPort2", result.ErrorMessage);
         }
+
+        [Fact]
+        internal async Task Test_CreateAsync_OnTemplateWithConditions()
+        {
+            using Bootstrapper bootstrapper = GetBootstrapper();
+            string templateLocation = GetTestTemplateLocation("TemplateWithConditions");
+            await InstallTemplateAsync(bootstrapper, templateLocation).ConfigureAwait(false);
+
+            string output = TestUtils.CreateTemporaryFolder();
+
+            IReadOnlyList<ITemplateMatchInfo> foundTemplates = await bootstrapper
+                .GetTemplatesAsync(new[] { WellKnownSearchFilters.NameFilter("TestAssets.TemplateWithConditions") })
+                .ConfigureAwait(false);
+
+            Dictionary<string, string?> parameters = new()
+            {
+                { "B", "true" },
+            };
+
+            ITemplateCreationResult result = await bootstrapper
+                .CreateAsync(foundTemplates[0].Info, "Test", output, parameters)
+                .ConfigureAwait(false);
+
+            Assert.Equal(CreationResultStatus.Success, result.Status);
+
+            string targetFile = Path.Combine(output, "test.ps1");
+            Assert.True(File.Exists(targetFile));
+            Assert.Equal(
+                $"# comment B true\r\nB true\r\ncommon text",
+                File.ReadAllText(targetFile).Trim());
```

Comment:
Could you clarify what you mean by " normalize newlines of both strings"?

> I'd probably normalize newlines of both strings prior comaprisons - just to avoid possible testbreaks caused by accidental newlines alteration of the test asset file (can be caused by rogue `.gitattributes` files or simply some refactoring changes)

Reply:
(discussed offline)

Btw. here is helper function: https://github.com/dotnet/templating/blob/main/test/Microsoft.TemplateEngine.TestHelper/StringExtensions.cs#L8